monkestation/code/datums/components/chain.dm

```diff
@@ -0,0 +1,108 @@
+
+/*
+This component attaches to things that'll follow another object
+Like connecting a person to a wheelchair so they get dragged behind it
+*/
+/datum/component/chain
+	//The thing the parent will follow
+	var/atom/movable/attachment_point
+
+	var/atom/movable/owner
+	var/max_dist
+	var/equal_force
+
+	var/datum/beam/tether_beam
+
+/datum/component/chain/Initialize(target, max_dist=3, equal_force = TRUE)
+	if(!ismovableatom(parent) || !ismovableatom(target))
+		return COMPONENT_INCOMPATIBLE
+
+	owner = parent
+	attachment_point = target
+	src.max_dist = max_dist
+	src.equal_force = equal_force
+
+	tether_beam = owner.Beam(target, "usb_cable_beam", 'icons/obj/wiremod.dmi')
```

Comment:
should probably use the maximum distance flag on this, unsure if it will help with the choppy beam, that will most likely require updating the beam on move

Reply:
try calling tether_beam.recalculate() on movement to get it to redraw nicer

Reply:
Tried this, it did make it better, it's jumping ahead instead of lagging behind, but it is a lot less choppy.


https://user-images.githubusercontent.com/99915170/220001946-982a3b0f-de11-4d76-ae5e-cfbe4439e44d.mp4

Reply:
thats because of glide, without glide it would look fine how you'd solve this would be to make a recalculate proc that animates the beam from its old location to its new location based on the parents glidesize

Reply:
Im having trouble animating it since 1 beam made up of a bunch of smaller ebeams. even if I did get it to animate I think it would still be scuffed if the parts dont move perfectly. do you know of anywhere I could look at that has beams with animated movement like how we want it?

Reply:
okay yea that won't work looking at beam code now, hm this will have to do until we get something better than current beam code